test/mlir/krnl/memory_pool.mlir

```diff
@@ -50,14 +47,10 @@ func @test_memory_pool(%arg0: tensor<10x10xf32>) -> tensor<10x10xf32> {
 
   /// Usage of the getref MemRef.
   // CHECK: [[MEM0:%.+]] = llvm.extractvalue [[MEMREF1_TMP7]][1] : !llvm.struct<(ptr<float>, ptr<float>, i64, array<2 x i64>, array<2 x i64>)>
-  // CHECK: [[CONST7:%.+]] = llvm.mlir.constant(0 : index) : !llvm.i64
-  // CHECK: [[CONST8:%.+]] = llvm.mlir.constant(10 : index) : !llvm.i64
-  // CHECK: [[MUL1:%.+]] = llvm.mul {{.*}}, [[CONST8]] : !llvm.i64
-  // CHECK: [[ADD1:%.+]] = llvm.add [[CONST7]], [[MUL1]] : !llvm.i64
-  // CHECK: [[CONST9:%.+]] = llvm.mlir.constant(1 : index) : !llvm.i64
-  // CHECK: [[MUL2:%.+]] = llvm.mul {{.*}}, [[CONST9]] : !llvm.i64
-  // CHECK: %[[ADD2:.+]] = llvm.add [[ADD1]], [[MUL2]] : !llvm.i64
-  // CHECK: llvm.getelementptr [[MEM0]][%[[ADD2]]] : (!llvm.ptr<float>, !llvm.i64) -> !llvm.ptr<float>
+  // CHECK: [[CONST7:%.+]] = llvm.mlir.constant(10 : index) : !llvm.i64
+  // CHECK: [[MUL1:%.+]] = llvm.mul {{.*}}, [[CONST7]] : !llvm.i64
+  // CHECK: %[[ADD1:.+]] = llvm.add [[MUL1]], {{.*}} : !llvm.i64
+  // CHECK: llvm.getelementptr [[MEM0]][%[[ADD1]]] : (!llvm.ptr<float>, !llvm.i64) -> !llvm.ptr<float>
```

Comment:
Looks like the offset computation is wrong. What led to this simplified computation?

Reply:
I believe this is due to the same folding changes as above. As far as I can tell all that's being removed is a multiply by 1 and an add of 0.